### PR TITLE
Titania decomp - 4 non-matching, 2 non-equiv (possibly equivalent)

### PR DESCRIPTION
**Pull request by inspectredc**
A lot more of a slog than I expected from this one

### CODE REVIEW

**Review by sonicdcer — APPROVED** [review]

File: src/overlays/ovl_i5/fox_ti.c [hunk 1]
```diff
@@ -1,138 +1,5375 @@
...
+
+    D_i5_801BBEF8 = SEGMENTED_TO_VIRTUAL(D_8000FC0);
+}
+
+bool func_i5_8018FC70(s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3f* rot, void* data) {
+    f32 sp24;
+
+    if (D_i5_801BBEF0[30] != 0) {
+        sp24 = (D_i5_801BBEF0[30] / 15.0f) * D_i5_801BBEF0[32];
+        rot->z += __sinf((D_i5_801BBEF0[30] / (f32) D_i5_801BBEF0[31]) * 360.0f * M_DTOR) * sp24;
```
Comment (petrie911) [inline src/overlays/ovl_i5/fox_ti.c:2256]:
```suggestion
        rot->z += SIN_DEG((D_i5_801BBEF0[30] / (f32) D_i5_801BBEF0[31]) * 360.0f) * sp24;
```

File: src/overlays/ovl_i5/fox_ti.c [hunk 2]
```diff
@@ -1,138 +1,5375 @@
...
+        }
+    }
+
+    for (i = 0; i < 4; i++) {
+        if (boss->swork[i + 5] > 0) {
+            boss->swork[i + 5]--;
+        }
+    }
+
+    for (i = 0; i < 0x22; i++) {
```
Comment (petrie911) [inline src/overlays/ovl_i5/fox_ti.c:3783]:
```suggestion
    for (i = 0; i < 34; i++) {
```
or perhaps array count

File: src/overlays/ovl_i5/fox_ti.c [hunk 3]
```diff
@@ -1,138 +1,5375 @@
...
+                    D_Timer_80161A60 = 8;
+                    Audio_PlaySfx(0x2940D09A, boss->sfxPos, 4, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+                } else {
+                    Audio_PlaySfx(0x2940802C, boss->sfxPos, 4, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+
+                    if (boss->swork[21] < 0xB) {
+                        D_i5_801BBEF0[1] = 15;
+                    } else if (boss->swork[21] < 0x15) {
+                        D_i5_801BBEF0[1] = 20;
+                    }
```
Comment (petrie911) [inline src/overlays/ovl_i5/fox_ti.c:3900]:
```suggestion
                    if (boss->swork[21] <= 10) {
                        D_i5_801BBEF0[1] = 15;
                    } else if (boss->swork[21] <= 20) {
                        D_i5_801BBEF0[1] = 20;
                    }
```

File: src/overlays/ovl_i5/fox_ti.c [hunk 4]
```diff
@@ -1,138 +1,5375 @@
...
+                        var_v1++;
+                    }
+                }
+                if ((var_v1 == 3) && (D_i5_801BBEF0[8] == 0)) {
+                    D_i5_801BBF00[39].unk_26 |= 0x11;
+                }
+                if (var_v1 == 4) {
+                    D_i5_801BBEF0[41] = 0;
+                    boss->swork[28] = 1;
+                    boss->swork[0x1B] = 0x50;
```
Comment (petrie911) [inline src/overlays/ovl_i5/fox_ti.c:3882]:
```suggestion
                    boss->swork[27] = 80;
```
other work indices should also be decimal.

File: src/overlays/ovl_i5/fox_ti.c [hunk 5]
```diff
@@ -1,138 +1,5375 @@
...
+                    boss->swork[35]++;
+                }
+            } else if ((gBossHealthBar > 0) && (D_i5_801B7904[boss->dmgPart] == 0x17) && (boss->swork[29] != 0) &&
```
Comment (petrie911) [inline src/overlays/ovl_i5/fox_ti.c:3885]:
```suggestion
            } else if ((gBossHealthBar > 0) && (D_i5_801B7904[boss->dmgPart] == 23) && (boss->swork[29] != 0) &&
```

File: src/overlays/ovl_i5/fox_ti.c [hunk 6]
```diff
@@ -1,138 +1,5375 @@
...
+                boss->swork[28]++;
+            }
+            break;
+        case 8:
+            Audio_PlaySfx(0x11033022, boss->sfxPos, 4, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+
+            for (i = 0; i < 0x21; i++) {
+                actor = func_800A3608(0xBD);
+                if (actor != NULL) {
+                    actor->state = 0x28;
```
Comment (petrie911) [inline src/overlays/ovl_i5/fox_ti.c:4053]:
```suggestion
                    actor->state = 40;
```

File: src/overlays/ovl_i5/fox_ti.c [hunk 7]
```diff
@@ -1,138 +1,5375 @@
...
+        boss->fwork[0x31] -= 360.0f;
+    }
+
+    while (boss->fwork[0x31] < 0.0f) {
+        boss->fwork[0x31] += 360.0f;
+    }
+
+    for (i = 0; i < 4; i++) {
+        if (boss->swork[i + 9] == 0) {
+            j = boss->swork[i + 0x11];
```
Comment (petrie911) [inline src/overlays/ovl_i5/fox_ti.c:4654]:
```suggestion
            j = boss->swork[17 + i];
```
when indexing work buffers like this, it's better to have the offset first.

File: src/overlays/ovl_i5/fox_ti.c [hunk 8]
```diff
@@ -1,138 +1,5375 @@
...
+        boss->fwork[0xF] = boss->fwork[9];
+    }
+    if ((boss->state >= 7) && (D_i5_801B7738[boss->state] > 0.0f)) {
+        // Can use temp here
+        boss->fwork[2] =
+            (((-Math_SmoothStepToF(&boss->obj.pos.z, gPlayer[0].pos.z - D_i5_801BBEF4[11], 0.1f, 14.0f, 0.001f) /
+               14.0f) *
+              1.2f) +
+             1.0f) *
+            2.0f;
```
Comment (petrie911) [inline src/overlays/ovl_i5/fox_ti.c:4746]: Can said temp use get the formatter to output something less crazy?

File: src/overlays/ovl_i5/fox_ti.c [hunk 9]
```diff
@@ -1,138 +1,5375 @@
...
+                            }
+                            D_i5_801BBEF0[42] = 1;
+                            break;
+                        default:
+                            break;
+                    }
+
+                    if (boss->swork[4] >= 4) {
+                        if (!(D_i5_801BBEF0[46] & 1)) {
+                            if ((gTeamShields[1] > 0) && (D_i5_801BBEF0[8] == 0) && (gRadioState == 0) &&
```
Comment (petrie911) [inline src/overlays/ovl_i5/fox_ti.c:4888]: A lot of these nested if statements could probably use `||` to reduce indentation level.

File: src/overlays/ovl_i5/fox_ti.c [hunk 10]
```diff
@@ -1,138 +1,5375 @@
...
+        }
+    }
+    boss->swork[31]++;
+}
+
+static f32 D_i5_801B8D5C[50] = { 0.0f,   0.03f, 0.06f,  0.12f, 0.18f,  0.25f, 0.425f, 0.6f, 0.775f, 0.95f,
+                                 1.125f, 1.3f,  1.475f, 1.65f, 1.825f, 2.0f,  1.8f,   1.6f, 1.4f,   1.2f,
+                                 1.0f,   0.8f,  0.6f,   0.4f,  0.2f,   0.0f,  0.8f,   1.6f, 2.4f,   3.2f,
+                                 4.0f,   3.4f,  3.2f,   3.0f,  2.8f,   2.4f,  2.2f,   2.0f, 1.8f,   1.6f,
+                                 1.4f,   1.2f,  1.0f,   0.8f,  0.6f,   0.4f,  0.2f,   0.0f, 0.0f,   0.0f };
```
Comment (petrie911) [inline src/overlays/ovl_i5/fox_ti.c:4957]:
```suggestion
                                 1.4f,   1.2f,  1.0f,   0.8f,  0.6f,   0.4f,  0.2f,   0.0f, 0.0f,   0.0f, };
```

File: src/overlays/ovl_i5/fox_ti.c [hunk 11]
```diff
@@ -1,138 +1,5375 @@
...
+            break;
+        case 120:
+            actor = func_800A3608(OBJ_ACTOR_189);
+            if (actor != NULL) {
+                if (Rand_ZeroOne() < 0.5f) {
+                    var_s1 = -1;
+                } else {
+                    var_s1 = 1;
+                }
+                actor->fwork[0] = (RAND_FLOAT(5.0f) + 5.0f) * (f32) var_s1;
```
Comment (petrie911) [inline src/overlays/ovl_i5/fox_ti.c:5308]:
```suggestion
                actor->fwork[0] = (RAND_FLOAT(5.0f) + 5.0f) * ((Rand_ZeroOne() < 0.5f) ? 1 : -1);
```
You don't have to make this change, but I'm curious if something like this still matches. If so, macroing up `((Rand_ZeroOne() < 0.5f) ? 1 : -1)` could be reasonable.

File: src/overlays/ovl_i5/fox_ti.c [hunk 12]
```diff
@@ -1,138 +1,5375 @@
...
+            break;
+        default:
+            break;
+    }
+}
+
+extern s16 D_i5_801BAA50[19];
+extern s16 D_i5_801BAA78[34];
+extern s16 D_i5_801BAABC[4][10];
+extern s16 D_i5_801BAB0C[4][10];
```
Comment (petrie911) [inline src/overlays/ovl_i5/fox_ti.c:3640]: setting these as const arrays should get them into rodata and let you remove the D symbol pragmas from the bottom.

**Comment by sonicdcer** [comment]
Outstanding work, you got one of the hardest files from this game my friend. Thank you for this contribution!

File: src/overlays/ovl_i5/fox_ti.c (hunk 11 above)
Comment (inspectredc) [reply src/overlays/ovl_i5/fox_ti.c:5308]: forgot to use that on this function ah, got a non-fake match now thanks

**Pull request merged**